### ISSUE TITLE
Support unix domain sockets in HttpClient

### ISSUE THREAD

**Issue by gregw** [eclipse/jetty.project#2014]
extend the support for unix domain sockets to HttpClient

**Comment by gregw** [comment]
The client still has regular unit test failures, so I have disabled the tests in the build for now and we need to review this implementation.

**Comment by olamy** [comment]
any logs?

**Comment by gregw** [comment]
I didn't capture them, but it produces verbose server and client dumps when it does fail, so it is a very long trace.   I think this is currently a low priority, so let's no look at this until 9.4.9 is out.

**Comment by gregw** [comment]
Ah here is the start of a test failure:
```
Running org.eclipse.jetty.http.client.HttpClientLoadTest.testConcurrent[transport: UNIX_SOCKET]()
2018-01-31 11:06:34.191:INFO:oejs.Server:main: jetty-9.4.z-SNAPSHOT; built: unknown; git: unknown; jvm 1.8.0_152-b16
2018-01-31 11:06:34.474:INFO:oejs.AbstractConnector:main: Started UnixSocketConnector@1cd072a9{HTTP/1.1,[http/1.1]}{/tmp/unix2656145385526675422.sock}
2018-01-31 11:06:34.474:INFO:oejs.Server:main: Started @668ms
2018-01-31 11:06:34.497:INFO:oejus.SslContextFactory:main: x509=X509@ba8d91c(mykey,h=[],w=[]) for SslContextFactory@7364985f[provider=null,keyStore=file:///usr/local/home/gregw/src/jetty-9.4.x/tests/test-http-client-transport/src/test/resources/keystore.jks,trustStore=file:///usr/local/home/gregw/src/jetty-9.4.x/tests/test-http-client-transport/src/test/resources/truststore.jks]
2018-01-31 11:06:34.706:WARN:oejh.HttpParser:client-44: Illegal character 0xA in state=RESPONSE_VERSION for buffer DirectByteBuffer@23d2d3c1[p=8,l=16384,c=16384,r=16376]={\r\n71A4\r\n<<<\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00>>>}
org.eclipse.jetty.client.HttpResponseException: HTTP protocol violation: bad response on HttpConnectionOverHTTP@16b972ac(l:null <-> r:null,closed=false)=>HttpChannelOverHTTP@53bbee1f(exchange=HttpExchange@49ca7582 req=TERMINATED/null@null res=PENDING/null@null)[send=HttpSenderOverHTTP@4720824d(req=QUEUED,snd=COMPLETED,failure=null)[HttpGenerator@6222f949{s=START}],recv=HttpReceiverOverHTTP@3f53cbb3(rsp=IDLE,failure=null)[HttpParser{s=CLOSE,0 of -1}]]<-UnixSocketEndPoint@33941150{null<->null,OPEN,fill=-,flush=-,to=1/0}{io=0/0,kio=0,kro=1}->HttpConnectionOverHTTP@16b972ac(l:null <-> r:null,closed=false)=>HttpChannelOverHTTP@53bbee1f(exchange=HttpExchange@49ca7582 req=TERMINATED/null@null res=PENDING/null@null)[send=HttpSenderOverHTTP@4720824d(req=QUEUED,snd=COMPLETED,failure=null)[HttpGenerator@6222f949{s=START}],recv=HttpReceiverOverHTTP@3f53cbb3(rsp=IDLE,failure=null)[HttpParser{s=CLOSE,0 of -1}]]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.badMessage(HttpReceiverOverHTTP.java:334)
	at org.eclipse.jetty.http.HttpParser.badMessage(HttpParser.java:1528)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1510)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:173)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:134)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:72)
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:133)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:155)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:242)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:384)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$1.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by sbordet** [comment]
I think we should wait for https://openjdk.java.net/jeps/380.

**Comment by sbordet** [comment]
Closed in favor of #6043.
